homeassistant/components/tplink/light.py

```diff
@@ -163,42 +130,45 @@ def available(self) -> bool:
     @property
     def device_state_attributes(self):
         """Return the state attributes of the device."""
-        return self._emeter_params
+        return self._light_state.emeter_params
 
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
-        if ATTR_BRIGHTNESS in kwargs:
-            brightness = int(kwargs[ATTR_BRIGHTNESS])
-        elif self._light_state.brightness is not None:
-            brightness = self._light_state.brightness
-        else:
-            brightness = 255
-
-        if ATTR_COLOR_TEMP in kwargs:
-            color_tmp = int(kwargs[ATTR_COLOR_TEMP])
-        else:
-            color_tmp = self._light_state.color_temp
-
-        if ATTR_HS_COLOR in kwargs:
-            # TP-Link requires integers.
-            hue_sat = tuple(int(val) for val in kwargs[ATTR_HS_COLOR])
-
-            # TP-Link cannot have both color temp and hue_sat
-            color_tmp = 0
-        else:
-            hue_sat = self._light_state.hs
-
-        await self._async_set_light_state_retry(
+        brightness = (
+            int(kwargs[ATTR_BRIGHTNESS])
+            if ATTR_BRIGHTNESS in kwargs
+            else self._light_state.brightness
+            if self._light_state.brightness is not None
+            else 255
```

Comment:
We shouldn't use inline if-else statements if they wrap multiple lines. It hurts readability.

Reply:
Yes, my plan is to merge the relevant commits from #33606 to this PR, which will also fix that. I'm just waiting for a test bulb to arrive, so that I can verify that things still work after I do that.